src/pepr/operator/controllers/network/authorizationPolicies.ts

```diff
@@ -0,0 +1,269 @@
+  }
+  return { source, ports };
+}
+
+/**
+ * Processes an Expose rule to extract its effective source and ports.
+ */
+function processExposeRule(rule: Expose): { source: Source; ports: string[] } {
+  const ports: string[] = [];
+  const effectivePort = rule.targetPort ?? rule.port;
+  if (effectivePort !== undefined) {
+    ports.push(effectivePort.toString());
+  }
+  const source =
+    rule.gateway === Gateway.Admin
+      ? { principals: [ADMIN_INGRESS] }
+      : { principals: [TENANT_INGRESS] };
+  return { source, ports };
+}
+
+/**
+ * Helper to determine if an object is empty.
+ */
+function isEmpty(obj: object): boolean {
+  return Object.keys(obj).length === 0;
+}
+
+/**
+ * Helper to build an AuthorizationPolicy from rule details.
+ * If the computed source is empty, the "from" field is omitted.
+ */
+function buildAuthPolicy(
+  pkgName: string,
+  pkgNamespace: string,
+  generation: string,
+  rule: Allow | Expose,
+  source: Source,
+  ports: string[],
+): AuthorizationPolicy {
+  const policyName = `protect-${pkgName}-${generateAuthPolName(rule)}`;
+  const ruleEntry: Rule = {};
+  if (!isEmpty(source)) {
+    ruleEntry.from = [{ source }];
+  }
+  if (ports.length > 0) {
+    ruleEntry.to = [{ operation: { ports } }];
+  }
+  return {
+    apiVersion: "security.istio.io/v1beta1",
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: policyName,
+      namespace: pkgNamespace,
+      labels: { "uds/package": pkgName, "uds/generation": generation },
+    },
+    spec: {
+      action: Action.Allow,
+      ...(rule.selector ? { selector: { matchLabels: rule.selector } } : {}),
+      rules: [ruleEntry],
+    },
+  };
+}
+
+/**
+ * Helper to generate a monitor AuthorizationPolicy.
+ */
+function buildMonitorAuthPolicy(
+  pkgName: string,
+  pkgNamespace: string,
+  generation: string,
+  monitor: Monitor,
+): AuthorizationPolicy {
+  const mSelector = monitor.podSelector ?? monitor.selector;
+  const monitorRule: Rule = {
+    from: [{ source: { namespaces: ["monitoring"] } }],
+    to: [{ operation: { ports: [monitor.targetPort.toString()] } }],
+  };
+  const monitorPolicyName = `protect-${pkgName}-monitor-${derivePolicyName(mSelector)}-${monitor.targetPort}`;
+  return {
+    apiVersion: "security.istio.io/v1beta1",
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: monitorPolicyName,
+      namespace: pkgNamespace,
+      labels: { "uds/package": pkgName, "uds/generation": generation },
+    },
+    spec: {
+      action: Action.Allow,
+      selector: { matchLabels: mSelector },
+      rules: [monitorRule],
+    },
+  };
+}
+
+/**
+ * Generate and apply Istio Authorization Policies for a given UDSPackage.
+ */
+export async function generateAuthorizationPolicies(
+  pkg: UDSPackage,
+  overrideNamespace?: string,
+): Promise<AuthorizationPolicy[]> {
+  const pkgName = pkg.metadata?.name ?? "unknown";
+  const pkgNamespace = pkg.metadata?.namespace ?? overrideNamespace ?? "default";
```

Comment:
It should be impossible to hit the fallback here (`default`) I think, but is there a reason that `overrideNamespace` isn't just directly used? I think we assume it exists for the netpols so thinking we could similar here?